src/main/java/uk/gov/hmcts/reform/sscs/service/SubmitAppealService.java

```diff
@@ -73,6 +78,10 @@ public Long submitAppeal(SyaCaseWrapper appeal, String userToken) {
 
     public Optional<SaveCaseResult> submitDraftAppeal(String oauth2Token, SyaCaseWrapper appeal) {
         appeal.setCaseType("draft");
+        if (!verifyTokenService.verifyTokenSignature(oauth2Token)) {
+            throw new InvalidSubscriptionTokenException(new Exception());
```

Comment:
Might be worth checking with Thor and Hem about how we want to handle this error scenario. Do you know what happens at the user level at the FE if this exception occurs? I just had a quick look, and it seems the BE will throw this exception resulting in an exception in AppInsight. And the FE we catching all errors so the user will not see any error and will continue as if nothing has happened whereas the draft is not being saved at all. This might confuse users that may think the draft was stored successfully but it was not.

Reply:
we will create a ticket to handle this scenario.